linux/ibus-keyman/src/engine.c

```diff
@@ -626,6 +680,28 @@ ibus_keyman_engine_process_key_event (IBusEngine     *engine,
     return TRUE;
  }
 
+static void
+ibus_keyman_engine_set_surrounding_text (IBusEngine *engine,
+                                            IBusText    *text,
+                                            guint       cursor_pos,
+                                            guint       anchor_pos)
+{
+    g_message("ibus_keyman_engine_set_surrounding_text");
+    parent_class->set_surrounding_text (engine, text, cursor_pos, anchor_pos);
+    reset_context(engine);
```

Comment:
Will this forward a corresponding reset of context to the KMX processor? That may end up losing markers (deadkeys) incorrectly?

Reply:
changed to checking if surrounding text has changed cf what is expected from KMX processor context before force setting it.
also changed to buffer multiple consecutive (i.e. with no BACK or EMIT in between) CHAR actions